Update readme.md (#1)
DevExpress-Examples/dashboard-for-winforms-diagnose-performance: readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T1085817)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->
# Dashboard for WinForms - Inspect the Dashboard Performance

The following example shows how to integrate the Dashboard Diagnostic Tool's functionality in the Dashboard Designer to allow users to inspect the dashboard's performance.

## Overview

> The **Inspect** button is inserted to the custom Ribbon's **Performance Diagnostics** page group. The button click starts a new diagnostic session. The Dashboard Diagnostic Tool monitors user actions to collects actions for each event of the event session. When you uncheck the button, the session ends and the resulting report is automatically saved to the XML file. You can open the report in the Diagnostic Tool UI.

Reviewer: ```suggestion
The **Inspect** button is inserted into the custom Ribbon's **Performance Diagnostics** page group. The button click starts a new diagnostic session. The Dashboard Diagnostic Tool monitors user actions to collects actions for each event of the event session. When you uncheck the button, the session ends and the resulting report is automatically saved to the XML file. You can open the report in the Diagnostic Tool UI.
```

> ![Dashboard Diagnostic Tool integrated to the Dashboard Designer](./images/dashboardMain.png)

Reviewer: ```suggestion
![Dashboard Diagnostic Tool integrated into the Dashboard Designer](./images/dashboardMain.png)

```

## Inspect the Dashboard Performance in the Dashboard Designer

1. Download the [Dashboard Diagnostic Tool](https://github.com/DevExpress-Examples/bi-dashboard-diagnosic-tool). 

2. Reference `DiagnosticTool.dll` and install the [Microsoft.Diagnostics.Tracing.TraceEvent](https://www.nuget.org/packages/Microsoft.Diagnostics.Tracing.TraceEvent/) package in your dashboard project. 

> 3. Create a custom button and insert it to the Ribbon.

Reviewer: ```suggestion
3. Create a custom button and insert it in the Ribbon.

```

4. Create a `DiagnosticController` object. 

> 5. Call the conroller's Start() and Stop() methods on button's click to run and finish the Dashboard Diagnostic Tool's session.

Reviewer: Note that "conroller's" is mispelled.
Also, what do you mean by "on button's click"? For example, do you mean:

`5. Call the controller's Start() and Stop() methods on a button click to run and finish the Dashboard Diagnostic Tool's session.`

or

`5. Call the controller's Start() and Stop() methods on each button click to run and finish the Dashboard Diagnostic Tool's session.`

Author: ```suggestion
5. Call the controller's Start() and Stop() methods on a button click to run and finish the Dashboard Diagnostic Tool's session.

```

> 6. Implement the `IFileController` interface and specify the output file path in the `TrySaveFile` method. Pass a new class instance that implements `IFileController` to the controller's contructor. 

Reviewer: 1. "contructor" is misspelled. Please change it to "constructor".

2. I don't think "implements `IFileController` _to_" is correct.
Do you mean "implements `IFileController` _in_ the controller's constructor"?

Author: ```suggestion
6. Implement the `IFileController` interface and specify the output file path in the `TrySaveFile` method. Pass a new class instance that implements `IFileController` in the controller's constructor. 

```
